bowling_game.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.push('Users/andrespacheco/Documents/andres-pacheco/bowling-game-kata-pacheco-andres')
+
+load 'frame.rb'
+load 'game_roll.rb'
+
+# bowling game logic class
+class BowlingGame
+  def initialize
+    @pins_down_collection = [10, 2, 3, 0, 0, 4, 6, 0, 0, 3, 4, 10, 10, 9, 3, 8, 1]
+  end
+
+  def start
+    @frames = []
+    game_roll = GameRoll.new
```

Comment:
frames and game_roll can be attribute_readers and can be used everywhere inside the class instead of global variables

Reply:
```ruby
class BowlingGame
  attr_reader :frames, :game_roll, :pins_down_collection
  def initialize
    @pins_down_collection = [10, 2, 3, 0, 0, 4, 6, 0, 0, 3, 4, 10, 10, 9, 3, 8, 1]
    @frames = []
    @game_roll = GameRoll.new
  end

  def start
    pins_down_collection.each do |pins_down|
      pins_down_collection_iterator(pins_down)
    end
  end

  def pins_down_collection_iterator(pins_down)
    game_roll.add(pins_down)
    frame = Frame.new

    if frames.count == 10
      strike_or_spare(frame, pins_down)
    elsif game_roll.strike? && frames.count <= 9
      frame.add(pins_down)
      frames << frame
      game_roll.reset
    elsif game_roll.completed? && frames.count <= 9
      frame.add(game_roll.first_turn_pins_down)
      frame.add(game_roll.second_turn_pins_down)
      frames << frame
      game_roll.reset
    end
  end

  def strike_or_spare(frame, pins_down)
    if frames.last.strike? || frames.last.spare?
      frame.add(pins_down)
      frames << frame
    end
  end

  def score
    score = 0
    bonus = 0

    frames.each.with_index do |frame, index|
      score += frame.turns.sum

      if index == 9 && (frame.strike? || frame.spare?)
        bonus += frames[index + 1].turns.first
      elsif frame.strike?
        if frames[index + 1].strike?
          bonus += frames[index + 1].turns.sum
          bonus += frames[index + 2].turns.first
        else
          bonus += frames[index + 1].turns.sum
        end
      elsif frame.spare?
        bonus += frames[index + 1].turns.first
      end

      score += bonus
      bonus = 0
    end

    score
  end
end
```